community/feature-status.md

```diff
@@ -0,0 +1,152 @@
+1. If for some reason an _Alpha_ feature fails to be promoted, it will
+eventually be removed from the code base.
+1. _Beta_ features can transition to _Stable_.
+1. If for some reason a _Beta_ feature fails to be promoted, it could
+eventually be removed from the code base followed from at least one
+release deprecation period.
+1. If a serious issue is identified in a stable feature that justifies
+it to be removed, it will follow at least two releases deprecation
+period.
+
+If a _Beta_ or _Stable_ feature is decided to be removed, it will
+respect a deprecation period. The information will be available in the
+release notes and in the official documentation for that given
+feature.
+
+The diagram bellow represents the possible transitions features can
+have:
+
+``` 
+                 ┌───────┐
+ New Feature'───►│ Alpha ├───► Removed
+                 └───┬───┘
+                     │
+                     │
+                     │
+                     ▼
+                  ┌──────┐
+ New Feature"────►│ Beta ├───► Removed
+                  └──┬───┘
+                     │
+                     │
+                     ▼
+                 ┌────────┐
+                 │ Stable ├──► Removed
+                 └────────┘
+```
+
+Promoting a feature is a responsibility shared between maintainers,
+contributors and the user community. Any of those parties can request
+a feature to be promoted. Ideally this can be done by creating an
+issue or a discussion in the Github repository of the project
+containing that feature. Promotion requests can be brought to
+maintainers attention in our weekly [Argo Contributors Office
+hours][1] meeting.
+
+## What it means for Argo users
+
+Feature status is a framework to provide transparency and allow
+setting expectations for Argo users. It should be used to support the
+decision making process about whether or not a given feature is
+feasible to be leveraged by a company or individual. Argo projects
+depend on the user base to provide valuable feedback and submit bugs
+with good information so features can be promoted to a more mature
+stage.
+
+## What it means for Argo contributors
+
+If you are contributing with new feature in one of the Argo projects,
+please classify it with one of the statuses described in this
+document. The current status should be indicated next to every feature
+in our official documentation. The status must also be informed in the
+title of the PR introducing the feature so it can be available in the
+release notes.
+
+The format of a feature status in the PR title should be:
+
+`feat: <title> (<status>)`
+
+Example:
+
+`feat: My awesome new feature description (Alpha)`
+
+The format of a feature status in the documentation should be:
+
+`Current Status: <status> (Since <version>)`
+
+Since relates to the version a feature was introduced, not the version
+it changed the status.
+
+Example:
+
+`## Server-Side Apply Current Status: Beta (Since v2.5.0)`
```

Comment:
When an alpha feature is getting promoted into beta, do we expect a contributor to make a PR titled "feat: my previously built cool feature (beta)" ?

Reply:
Not necessarily. I needs to be updated is the page where the feature is documented to point to the current status.

Reply:
@sbose78 On a second thought, a PR will have to be created anyways to update at least the documentation. I think we should have the PR title following the pre-defined format in this case as well. Updated the docs. 
PTAL

Reply:
Thank you @leoluz , looks great.